### ISSUE TITLE
[Forge 1.20.1, 47.2.17] Item in hotbar slot #1 gets permanently replaced/deleted by mainhand slot on death.

### ISSUE THREAD

**Issue by gabrielgarethfoo** [CalaMariGold/ConfigurableDeath#2]
Can be reproduced by leaving any item in hotbar slot 1, and dying while another hotbar slot is actively selected.

Item in hotbar slot 1 gets deleted/replaced regardless of whether there are any other items in other hotbar slots. Hotbar slot 1 will get replaced by the currently selected mainhand item upon death. Dying with an empty hand will also cause hotbar slot 1 to get removed.

**Modlist:**
- Minecraft 1.20.1
- Forge 47.2.17
- configurabledeath1_20_1-1.0.0.jar

**Config settings used in configurabledeath-server.toml:**

```
[itemDrops]
	#Inventory drop settings
	#
	#Should players keep their inventory on death?
	keepInventory = false
	#Should players keep their armor on death?
	keepArmor = true
	#Should players keep their non-mainhand hotbar items on death?
	keepHotbar = true
	#Should players keep their mainhand item on death?
	keepMainhand = true
	#Should players keep their offhand item on death?
	keepOffhand = true
	#Set to true to keep main inventory (non-equipped non-hotbar) items on death
	keepMainInventoryOnDeath = false

[durability]
	#Durability loss settings
	#
	#Percent of durability lost on death for kept items
	#Range: 0.0 ~ 1.0
	durabilityLossOnKeptItems = 0.1
	#Percent of durability lost on death for drops
	#Range: 0.0 ~ 1.0
	durabilityLossOnDrops = 0.0

[experience]
	#Experience settings
	#
	#Enable experience settings
	enableExperienceModule = true
	#Percent of experience dropped on death
	#Range: 0.0 ~ 1.0
	droppedXPPercent = 0.5
	#Percent of dropped experience that can be recovered
	#Range: 0.0 ~ 1.0
	recoverableXPPercent = 0.8

[hunger]
	#Hunger settings
	#
	#Set to true to retain food level on death
	keepFoodLevel = true
	#Highest amount of food level you can respawn with
	#Range: 0 ~ 20
	maxFoodLevel = 20
	#Lowest amount of food level you can respawn with
	#Range: 0 ~ 20
	minFoodLevel = 20
	#Set to true to retain saturation on death
	keepSaturation = false
```

**Comment by CalaMariGold** [comment]
Ty for the detailed report!

**Comment by CalaMariGold** [comment]
fixed in 1.0.1
